store/iavl/store.go

```diff
@@ -82,18 +112,36 @@ func (st *Store) GetImmutable(version int64) (*Store, error) {
 		return nil, err
 	}
 
-	return &Store{tree: &immutableTree{iTree}}, nil
+	return &Store{
+		tree:    &immutableTree{iTree},
+		pruning: st.pruning,
+	}, nil
 }
 
-// Implements Committer.
+// Commit commits the current store state and returns a CommitID with the new
+// version and hash.
 func (st *Store) Commit() types.CommitID {
-	// Save a new version.
 	hash, version, err := st.tree.SaveVersion()
 	if err != nil {
 		// TODO: Do we want to extend Commit to allow returning errors?
 		panic(err)
 	}
 
+	// If the version we saved got flushed to disk, check if previous flushed
+	// version should be deleted.
+	if st.pruning.FlushVersion(version) {
+		previous := version - st.pruning.KeepEvery
+
+		// Previous flushed version should only be pruned if the previous version is
+		// not a snapshot version OR if snapshotting is disabled (SnapshotEvery == 0).
+		if previous != 0 && !st.pruning.SnapshotVersion(previous) {
```

Comment:
Maybe `previous > 0` is "safer".
Though I cannot think of a scenario when this could be negative, still better safe